### DISCUSSION TITLE
single identity for multiple addresses for MC, NomadNet and Sb. Is it feasible?

### DISCUSSION THREAD

**Discussion by EthicalObligation** [markqvist/Reticulum#598] [Help & Questions] [answered]
I have a single identity. I want to make more than one address for different applications. Sideband, MeshChat and NomadNet for example. Is it feasible?

**Comment by EthicalObligation** [comment]
There are some discussions that points out it's better to create an identity for each device. I suppose I want to know if something similar to DNS routing to ip addresses is possible.

**Comment by EthicalObligation** [comment]
For those who want to know so far I have been using rnsid with  -a something.program to make something similar. I am not sure this is the proper way of doing things.

**Comment by faragher** [comment, marked as answer]
If you use the same identity for the same destination on different machines, you're asking for trouble. From a TCP/IP perspective, think about if DNS pointed to three different IPs. It would cause no end of trouble, and it's not done. You can use a reverse proxy, but that's a server to handle links to multiple systems.

If you shut down the other systems prior to starting another, you would have less problems, but the network would route towards the last announce it saw. Not the last that was sent, the last it saw. If you were attached to the same node every time, only that node would need new pathing, but that doesn't ensure that the currently running system is the one that has a path. 

And to be clear, the messages only exist on the machine that received them, there is no possibility of multiple access. By jumping across systems, no data is shared and messages can be missed simply by virtue of switching back and forth. It will also break links in a way that's confusing for the other end.

So unless you are okay with checking each different software whenever you wanted to check for a message, it's not going to do what you want.

This is why I made POPR, which is a specification, not a software. You (or someone else) would need to write a server and client app, and it would have to run on a machine you trusted, as the messages would be stored unencrypted, but that's really the only way to do a multiple-access E2EE system: the *server* needs to be the recipient and the users need to log into the server independently and request the information. If that sounds similar to e-mail, in particular POP3 servers, that's deliberate. It solves a similar problem.

Again, it's a specification, not a software, but for reference: https://github.com/faragher/POPR

**Reply by EthicalObligation** [reply]
> If you use the same identity for the same destination on different machines, you're asking for trouble. From a TCP/IP perspective, think about if DNS pointed to three different IPs. It would cause no end of trouble, and it's not done. You can use a reverse proxy, but that's a server to handle links to multiple systems.
> 
> If you shut down the other systems prior to starting another, you would have less problems, but the network would route towards the last announce it saw. Not the last that was sent, the last it saw. If you were attached to the same node every time, only that node would need new pathing, but that doesn't ensure that the currently running system is the one that has a path.
> 
> And to be clear, the messages only exist on the machine that received them, there is no possibility of multiple access. By jumping across systems, no data is shared and messages can be missed simply by virtue of switching back and forth. It will also break links in a way that's confusing for the other end.
> 
> So unless you are okay with checking each different software whenever you wanted to check for a message, it's not going to do what you want.
> 
> This is why I made POPR, which is a specification, not a software. You (or someone else) would need to write a server and client app, and it would have to run on a machine you trusted, as the messages would be stored unencrypted, but that's really the only way to do a multiple-access E2EE system: the _server_ needs to be the recipient and the users need to log into the server independently and request the information. If that sounds similar to e-mail, in particular POP3 servers, that's deliberate. It solves a similar problem.
> 
> Again, it's a specification, not a software, but for reference: https://github.com/faragher/POPR

An exceptionally well written explanation! Thank you so much!
